src/main/java/org/broadinstitute/hellbender/tools/picard/sam/FilterReads.java

```diff
@@ -112,18 +112,17 @@ private void filterReads(final FilteringIterator filteringIterator) {
      *                     containing read names
      */
     private void writeReadsFile(final File samOrBamFile) throws IOException {
-        final SamReader reader = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(samOrBamFile);
-        final File readsFile =
-                new File(OUTPUT.getParentFile(), IOUtil.basename(samOrBamFile) + ".reads");
-        IOUtil.assertFileIsWritable(readsFile);
-        final BufferedWriter bw = IOUtil.openFileForBufferedWriting(readsFile, false);
-
-        for (final SAMRecord rec : reader) {
-            bw.write(rec.toString() + "\n");
+        File readsFile = null;
+        try (final SamReader reader = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(samOrBamFile)) {
+            readsFile = new File(OUTPUT.getParentFile(), IOUtil.basename(samOrBamFile) + ".reads");
+            IOUtil.assertFileIsWritable(readsFile);
+            try (final BufferedWriter bw = IOUtil.openFileForBufferedWriting(readsFile, false)) {
+
+                for (final SAMRecord rec : reader) {
+                    bw.write(rec.toString() + "\n");
+                }
+            }
         }
-
-        bw.close();
-        reader.close();
         IOUtil.assertFileIsReadable(readsFile);
```

Comment:
this line needs a null check for readsFile (in case an exception happens before readsFile is initialized to a file object)